packages/cli/src/UserAuthentication/methods/ldap.ts

```diff
@@ -0,0 +1,70 @@
+import { InternalHooksManager } from '@/InternalHooksManager';
+import { SignInType } from '@/Ldap/constants';
+import {
+	createLdapUserOnLocalDb,
+	findAndAuthenticateLdapUser,
+	getLdapConfig,
+	getLdapUserRole,
+	getUserByEmail,
+	getUserByLdapId,
+	isInstanceOwner,
+	isLdapDisabled,
+	mapLdapAttributesToDb,
+	transformEmailUserToLdapUser,
+	updateLdapUserOnLocalDb,
+} from '@/Ldap/helpers';
+import type { User } from '@db/entities/User';
+
+export const handleLdapLogin = async (
+	email: string,
+	password: string,
+): Promise<User | undefined> => {
+	if (isLdapDisabled()) return undefined;
+
+	if (await isInstanceOwner(email)) return undefined;
+
+	const ldapConfig = await getLdapConfig();
+
+	if (!ldapConfig.loginEnabled) return undefined;
+
+	const { loginIdAttribute, userFilter } = ldapConfig;
+
+	const ldapUser = await findAndAuthenticateLdapUser(email, password, loginIdAttribute, userFilter);
+
+	if (!ldapUser) return undefined;
+
+	const ldapAttributesValues = mapLdapAttributesToDb(ldapUser, ldapConfig);
+
+	const { ldapId: ldapIdAttributeValue, email: emailAttributeValue } = ldapAttributesValues;
```

Comment:
is there a scenario where `emailAttributeValue` wouldn't be the same as the `email` that's passed in to this function?

Reply:
Yes, it depends on what the login attribute is in the configuration. For example:

LDAP db record:

uid: ricardoe105
mail: ricardoespinoza105@gmail.com
sn: espinoza
givenName: givenName

If the login ID in the configuration is **mail**, then the email parameter will be the value **ricardoespinoza105@gmail.com**

if the login ID in the configuration is **uid** then the email parameter will take the value **ricardoe105**

I will change the name to login id for better readability. Also, I will revert the change to check whether it's the instance owner, as this only makes sense when the login ID is an email. It's ok to let the search fail and then check in handleEmailLogin again.